src/pages/ShareCodePage.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {View, ScrollView} from 'react-native';
+import ScreenWrapper from '../components/ScreenWrapper';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import QRShareWithDownload from '../components/QRShare/QRShareWithDownload';
+import compose from '../libs/compose';
+import reportPropTypes from './reportPropTypes';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../components/withCurrentUserPersonalDetails';
+import styles from '../styles/styles';
+import roomAvatar from '../../assets/images/avatars/room.png';
+import * as ReportUtils from '../libs/ReportUtils';
+import MenuItem from '../components/MenuItem';
+import Clipboard from '../libs/Clipboard';
+import * as Expensicons from '../components/Icon/Expensicons';
+import getPlatform from '../libs/getPlatform';
+import CONST from '../CONST';
+
+const propTypes = {
+    /** The report currently being looked at */
+    report: reportPropTypes,
+
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+const defaultProps = {
+    report: undefined,
+    ...withCurrentUserPersonalDetailsDefaultProps,
+};
+
+// eslint-disable-next-line react/prefer-stateless-function
+class ShareCodePage extends React.Component {
+    qrCodeRef = React.createRef();
+
+    render() {
+        const isReport = this.props.report != null && this.props.report.reportID != null;
+
+        const url = isReport ? `${CONST.NEW_EXPENSIFY_URL}r/${this.props.report.reportID}` : `${CONST.NEW_EXPENSIFY_URL}details?login=${this.props.session.email}`;
```

Comment:
👋 Heads up that this has caused a couple of regressions: https://github.com/Expensify/App/issues/19076 https://github.com/Expensify/App/issues/19062 https://github.com/Expensify/App/issues/19344 https://github.com/Expensify/App/issues/19039
All of those were caused by `this.props.session.email` not being encoded and some characters being lost, which made the link invalid.
We resolved this by encoding `this.props.session.email` with `encodeURIComponent`

Reply:
We used a fixed `CONST.NEW_EXPENSIFY_URL` here, which caused an [issue](https://github.com/Expensify/App/issues/19464#issuecomment-1620718282) as it didn't set the URL based on the environment.